tests/io/v3/vm/tags/test_tags_api.py

```diff
@@ -0,0 +1,830 @@
+        'tags': tag_ids
+    }
+
+    # Let's create sample response for api endpoint
+    test_response: dict = {
+        'job_id': '62210d02a7056d0297f50a8ddfbd549eaef1d0bc94e1ea3fad09'
+    }
+
+    # Let's register the response to api endpoint
+    responses.add(
+        responses.POST,
+        re.compile(f'{BASE_URL}/assets/assignments'),
+        match=[responses.matchers.json_params_matcher(payload)],
+        json=test_response
+    )
+
+    res = api.v3.vm.tags.assign(
+        assets=asset_ids,
+        tags=tag_ids
+    )
+
+    assert isinstance(res, str)
+    assert test_response['job_id'] == res
+
+
+@responses.activate
+def test_unassign(api):
+    '''
+    Test case for tags unassign method
+    '''
+    asset_ids: list = ['60bfce80-f695-4ed0-bd47-ec9e5b2946e5']
+    tag_ids: list = ['54595f6e-648c-4570-b967-9b4e6a947634']
+
+    # Let's create sample payload for api endpoint
+    payload: dict = {
+        'action': 'remove',
+        'assets': asset_ids,
+        'tags': tag_ids
+    }
+
+    # Let's create sample response for api endpoint
+    test_response: dict = {
+        'job_id': '62210d02a7056d0297f50a8ddfbd549eaef1d0bc94e1ea3fad09'
+    }
+
+    # Let's register the response for api endpoint
+    responses.add(
+        responses.POST,
+        re.compile(f'{BASE_URL}/assets/assignments'),
+        match=[responses.matchers.json_params_matcher(payload)],
+        json=test_response
+    )
+
+    res = api.v3.vm.tags.unassign(
+        assets=asset_ids,
+        tags=tag_ids
+    )
+
+    assert isinstance(res, str)
+    assert test_response['job_id'] == res
+
+
+@responses.activate
+def test_search(api):
+    '''
+    Test case for tags search method
+    '''
+    # Let's create sample response for api endpoint
+    test_response: dict = {
+        'values': [
+            {
+                'id': 'd3786a2c-1c0c-4b5d-8b38-c9fbf7db9a04',
+                'created_at': '2021-10-19T12:06:17.529Z',
+                'created_by': 'user@tenable.com',
+                'updated_at': '2021-10-19T12:06:17.529Z',
+                'updated_by': 'user@tenable.com',
+                'category_id': '36c260ae-e3ad-4e59-afba-e65d9d436d3d',
+                'value': 'Q7',
+                'type': 'static',
+                'category_name': 'Audi'
+            },
+            {
+                'id': '27d6612c-3356-4023-9815-cbf5dc298403',
+                'created_at': '2021-10-22T10:13:34.111Z',
+                'created_by': 'szaros+us-2a@tenable.com',
+                'updated_at': '2021-10-22T10:13:34.111Z',
+                'updated_by': 'szaros+us-2a@tenable.com',
+                'category_id': '4f3bd5fb-9f99-49c7-8deb-89f7596674d0',
+                'value': 'First Value',
+                'description': '',
+                'type': 'dynamic',
+                'category_name': 'Demo Tag',
+                'category_description': ''
+            }
+        ],
+        'pagination': {'total': 2, 'next': 'nextToken'}
```

Comment:
**nextToken** is not needed as total records are 2 and they are already part of response.